Source/CardStackController.m

```diff
@@ -154,151 +168,127 @@ - (UIFont *)titleFont {
             }
-            if (completion) {
-                completion();
+            [self updateCardLocationsAnimatedWithVerticalVelocity:velocity.y];
+        } else if (card.tag == self.cards.count - 1) {
+            if (velocity.y < 0.0f &&
+                fabs(velocity.y) > CardStackVerticalVelocityLimitWhenMakingCardCurrent) {
+                // apply animation only to the card being moved (the rest remain stationary)
+                POPSpringAnimation *springAnimation = [POPSpringAnimation animationWithPropertyNamed:kPOPViewFrame];
+                CGRect frame = [self frameForCardAtIndex:card.tag];
+                frame.origin.y = 0;
+                springAnimation.toValue = [NSValue valueWithCGRect:frame];
+                springAnimation.springBounciness = 8;
+                springAnimation.velocity = [NSValue valueWithCGRect:CGRectMake(0, velocity.y, 0, 0)];
+                springAnimation.completionBlock = ^(POPAnimation *anim, BOOL finished) {
+                    self.isOpen = NO;
+                    self.currentCardIndex = self.cards.count - 1;
+                    [self updateCardScales];
+                    [self updateCardLocations];
+                };
+                [card pop_addAnimation:springAnimation forKey:@"frame"];
+            } else {
+                [self updateCardLocationsAnimated:YES];
             }
-        }];
-    } else {
-        [self updateCardLocations];
-        if ([self.delegate respondsToSelector:@selector(cardStackControllerDidOpen:)]) {
-            [self.delegate cardStackControllerDidOpen:self];
         }
-        if (completion) {
-            completion();
+    } else if (self.panType == PTRemove) {
+        if (card.tag > self.currentCardIndex) {
+            return;
         }
-    }
-}
 
-- (void)closeStackAnimated:(BOOL)animated
-            withCompletion:(void(^)())completion {
-    if (self.isAnimating) {
-        return;
-    }
-
-    self.isOpen = NO;
-    if ([self.delegate respondsToSelector:@selector(cardStackControllerWillClose:)]) {
-        [self.delegate cardStackControllerWillClose:self];
-    }
-
-    if (animated) {
-        self.isAnimating = YES;
-        [UIView animateWithDuration:0.5 animations:^{
-            [self updateCardLocations];
-        } completion:^(BOOL finished) {
-            self.isAnimating = NO;
-            if ([self.delegate respondsToSelector:@selector(cardStackControllerDidClose:)]) {
-                [self.delegate cardStackControllerDidClose:self];
-            }
-            if (completion) {
-                completion();
-            }
-        }];
-    } else {
-        [self updateCardLocations];
-        if ([self.delegate respondsToSelector:@selector(cardStackControllerDidClose:)]) {
-            [self.delegate cardStackControllerDidClose:self];
-        }
-        if (completion) {
-            completion();
+        if (velocity.x < CardStackVerticalVelocityLimitWhenRemovingCard || self.cards.count < 2) {
```

Comment:
Would be better if you named this `if` instead of adding comments bellow

```objc
BOOL thisShouldBeDoneBecause = (....);
if (thisShouldBeDoneBecause) {
    // ...
} else {
    // ...
}
```